### PR TITLE
pcm: init at 201710

### PR DESCRIPTION
**Pull request by roosemberth**
Signed-off-by: Roosembert Palacios <roosembert.palacios@epfl.ch>

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/pcm/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name = "pcm-${version}";
+
+  src = fetchFromGitHub {
+    owner = "opcm";
+    repo = "pcm";
+    rev = "${version}";
+    sha256 = "02rq8739zwwbfrhagvcgf6qpmnswxl9b0qsld26rg6zp91v2npbj";
+  };
+
+  nativeBuildInputs = [ ];
```
Comment (nlewo) [inline pkgs/os-specific/linux/pcm/default.nix]: Could you remove this please?

File: pkgs/os-specific/linux/pcm/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  buildInputs = [ ];
```
Comment (nlewo) [inline pkgs/os-specific/linux/pcm/default.nix]: Same here

File: pkgs/os-specific/linux/pcm/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  buildPhase = ''
```
Comment (nlewo) [inline pkgs/os-specific/linux/pcm/default.nix]: I think this is the default behavior of the build phase. So you could also remove this.

File: pkgs/os-specific/linux/pcm/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    cp pcm*.x $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Processor counter monitor";
+    homepage = https://www.intel.com/software/pcm;
+    license = licenses.bsd3;
+    platforms = platforms.linux;
```
Comment (nlewo) [inline pkgs/os-specific/linux/pcm/default.nix]: Would you like to become the maintainer of this package? In this case, could you add yourself in the maintainers list?

**Review by nlewo — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/pcm/default.nix (hunk 4 above)
Comment (roosemberth) [reply pkgs/os-specific/linux/pcm/default.nix]: Done :)

**Comment by roosemberth** [comment]
Why is there a xfce label in this PR?

**Comment by nlewo** [comment]
@roosemberth you pushed commits that are not yours:) I don't know exactly what you did. But it sounds like a wrong rebasing.
I think this explain why this PR has a `xfce` label.

**Comment by roosemberth** [comment]
My bad, properly rebased.

**Comment by nlewo** [comment]
@GrahamcOfBorg build pcm

File: pkgs/os-specific/linux/pcm/default.nix [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, callPackage, fetchFromGitHub }:
```
Comment (nlewo) [inline pkgs/os-specific/linux/pcm/default.nix]: callPackage argument is useless.

**Comment by nlewo** [comment]
Could you please either fix the build for aarch64 or restrict this package to x86 only?

**Comment by roosemberth** [comment]
FYI, it's expected it won't build in other architectures than x86_64, it's an intel cpu tool

**Comment by roosemberth** [comment]
Fixed: restrict arch and remove useless argument

**Comment by nlewo** [comment]
ahh, i'm feeling a bit stupid with aarch64 :)
Thanks for your contrib!

**Pull request merged**